From f079e5f73032b58caef775a101a141f45fed0915 Mon Sep 17 00:00:00 2001
From: Alexey Dobriyan <adobriyan@gmail.com>
Date: Thu, 16 Jun 2011 11:01:34 +0000
Subject: net: remove mm.h inclusion from netdevice.h

Remove linux/mm.h inclusion from netdevice.h -- it's unused (I've checked manually).

To prevent mm.h inclusion via other channels also extract "enum dma_data_direction"
definition into separate header. This tiny piece is what gluing netdevice.h with mm.h
via "netdevice.h => dmaengine.h => dma-mapping.h => scatterlist.h => mm.h".
Removal of mm.h from scatterlist.h was tried and was found not feasible
on most archs, so the link was cutoff earlier.

Hope people are OK with tiny include file.

Note, that mm_types.h is still dragged in, but it is a separate story.

Signed-off-by: Alexey Dobriyan <adobriyan@gmail.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit b7f080cfe223b3b7424872639d153695615a9255)

Conflicts:

	arch/arm/mach-davinci/board-mityomapl138.c
	arch/arm/mach-davinci/dm646x.c
	arch/arm/mach-davinci/pm.c
	arch/arm/mach-imx/dma-v1.c
	arch/arm/mach-imx/mach-mx31_3ds.c
	arch/arm/mach-iop13xx/setup.c
	arch/arm/mach-mxs/devices/platform-auart.c
	arch/arm/mach-mxs/devices/platform-dma.c
	arch/arm/mach-mxs/devices/platform-fec.c
	arch/arm/plat-mxc/devices/platform-fec.c
	arch/arm/plat-mxc/devices/platform-fsl-usb2-udc.c
	arch/arm/plat-mxc/devices/platform-imx-fb.c
	arch/arm/plat-mxc/devices/platform-ipu-core.c
	arch/arm/plat-mxc/devices/platform-mxc-ehci.c
	arch/arm/plat-mxc/devices/platform-mxc-mmc.c
	arch/arm/plat-nomadik/include/plat/ste_dma40.h
	arch/x86/kernel/tboot.c
	crypto/async_tx/raid6test.c
	drivers/dma/coh901318.c
	drivers/dma/dmaengine.c
	drivers/dma/dmatest.c
	drivers/dma/ipu/ipu_idmac.c
	drivers/dma/ste_dma40.c
	drivers/media/dvb/mantis/mantis_ca.c
	drivers/media/dvb/mantis/mantis_evm.c
	drivers/media/dvb/mantis/mantis_hif.c
	drivers/media/dvb/mantis/mantis_ioc.c
	drivers/media/dvb/mantis/mantis_pcmcia.c
	drivers/media/dvb/mantis/mantis_uart.c
	drivers/media/dvb/mantis/mantis_vp1034.c
	drivers/mtd/nand/atmel_nand.c
	drivers/net/arm/ks8695net.c
	drivers/net/bnx2x/bnx2x.h
	drivers/net/can/janz-ican3.c
	drivers/net/can/softing/softing_fw.c
	drivers/net/can/softing/softing_main.c
	drivers/net/ethoc.c
	drivers/net/fec_mpc52xx.c
	drivers/net/greth.c
	drivers/net/irda/pxaficp_ir.c
	drivers/net/ks8851_mll.c
	drivers/net/sgiseeq.c
	drivers/net/stmmac/dwmac1000_core.c
	drivers/net/stmmac/dwmac1000_dma.c
	drivers/net/stmmac/dwmac100_core.c
	drivers/net/stmmac/dwmac100_dma.c
	drivers/net/stmmac/stmmac_ethtool.c
	drivers/net/stmmac/stmmac_mdio.c
	drivers/net/usb/cdc-phonet.c
	drivers/net/vxge/vxge-config.h
	drivers/net/wireless/ath/ath5k/base.c
	drivers/net/wireless/ath/ath9k/beacon.c
	drivers/net/wireless/ath/ath9k/init.c
	drivers/net/wireless/ath/ath9k/recv.c
	drivers/net/wireless/ath/ath9k/xmit.c
	drivers/staging/pohmelfs/crypto.c
	drivers/tty/serial/ifx6x60.c
	drivers/usb/gadget/f_phonet.c
	include/crypto/if_alg.h
	include/linux/dma-direction.h
	include/linux/dma-mapping.h
	include/linux/dmaengine.h
	include/linux/netdevice.h
	net/sched/sch_netem.c
	security/apparmor/lib.c

Signed-off-by: Simon Horman <horms@verge.net.au>
---
 drivers/mmc/host/tmio_mmc_dma.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/mmc/host/tmio_mmc_dma.c b/drivers/mmc/host/tmio_mmc_dma.c
index 2aa616d..86f259c 100644
--- a/drivers/mmc/host/tmio_mmc_dma.c
+++ b/drivers/mmc/host/tmio_mmc_dma.c
@@ -11,6 +11,7 @@
  */
 
 #include <linux/device.h>
+#include <linux/dma-mapping.h>
 #include <linux/dmaengine.h>
 #include <linux/mfd/tmio.h>
 #include <linux/mmc/host.h>
-- 
1.7.10.2.565.gbd578b5

